definitions/npm/react-redux_v5.x.x/flow_v0.62.x-/test_connect.js

```diff
@@ -0,0 +1,179 @@
+// @flow
+import React from "react";
+import { connect } from "react-redux";
+
+function testPassingPropsToConnectedComponent() {
+  type Props = {passtrough: number, fromStateToProps: string};
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passtrough} {this.props.fromStateToProps}</div>;
+    }
+  }
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: state.a
```

Comment:
`state.a` is number, `forMapStateToProps ` is string

Reply:
that's very true, thanks for pointing out! This had totally slipped trough my eyes. Seems that the inference chain gets broken here. Same with props. If `state.a` is replaced with e.g `'abc'`, tests starts to fail like they should.

This is one of the worst parts of Flow: It's really hard to know when Flow infers something as `any`

Interestingly, Flow versions prior to 0.57.1 catches that error, so it's clearly a regression in Flow.

Reply:
wrote a bug report about this: https://github.com/facebook/flow/issues/5792